plugins/out_es/es_bulk.c

```diff
@@ -77,13 +78,14 @@ int es_bulk_append(struct es_bulk *bulk, char *index, int i_len,
          *    1. rest of msgpack data size
          *    2. ratio from bulk json size and processed msgpack size.
          */
+        append_size = required - available;
         if (converted_size == 0) {
             /* converted_size = 0 causes div/0 */
             flb_debug("[out_es] converted_size is 0");
-            append_size = required - available;
         } else {
-            append_size = (whole_size - converted_size) /* rest of size to convert */
-                        * (bulk->size / converted_size); /* = json size / msgpack size */
+            int remaining_size = ceil((whole_size - converted_size) /* rest of size to convert */
```

Comment:
please move variable declaration to the top of the function

Reply:
@edsiper done